x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/timeline/timeline.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import React, { FunctionComponent, useMemo } from 'react';
+import {
+  EuiText,
+  EuiIcon,
+  EuiIconProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiIconTip,
+} from '@elastic/eui';
+
+import { PhasesExceptDelete } from '../../../../../../common/types';
+import { useFormData } from '../../../../../shared_imports';
+
+import { FormInternal } from '../../types';
+
+import {
+  calculateRelativeTimingMs,
+  normalizeTimingsToHumanReadable,
+  PhaseAgeInMilliseconds,
+} from '../../lib';
+
+import './_timeline.scss';
+import InfinityIconSvg from './infinity_icon.svg';
+
+const InfinityIcon: FunctionComponent<Omit<EuiIconProps, 'type'>> = (props) => (
+  <EuiIcon type={InfinityIconSvg} {...props} />
+);
+
+const toPercent = (n: number, total: number) => (n / total) * 100;
+
+const msTimeToOverallPercent = (ms: number, totalMs: number) => {
+  if (!isFinite(ms)) {
+    return 100;
+  }
+  if (totalMs === 0) {
+    return 100;
+  }
+  return toPercent(ms, totalMs);
+};
+
+/**
+ * Each phase, if active, should have a minimum width it occupies. The higher this
+ * base amount, the smaller the variance in phase size in the timeline. This functions
+ * as a min-width constraint.
+ */
+const SCORE_BUFFER_AMOUNT = 50;
+
+const i18nTexts = {
+  hotPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.hotPhaseSectionTitle', {
+    defaultMessage: 'Hot phase',
+  }),
+  warmPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.warmPhaseSectionTitle', {
+    defaultMessage: 'Warm phase',
+  }),
+  coldPhase: i18n.translate('xpack.indexLifecycleMgmt.timeline.coldPhaseSectionTitle', {
+    defaultMessage: 'Cold phase',
+  }),
+  deleteIcon: {
+    toolTipContent: i18n.translate('xpack.indexLifecycleMgmt.timeline.delteIconToolTipContent', {
+      defaultMessage: 'Data will be deleted at the end of this policy',
```

Comment:
> Data will be deleted at the end of this policy

While the current text is okay, I don't think `policy` is the right word here. I also think we should say `index` in place of `data`. Otherwise users may assume we're deleting individual documents.

Here are a couple alternatives in order of preference:

1. The index will be deleted at the end of the lifecycle
2. The index will be deleted at the end of this phase

Reply:
+1 to @debadair's points and suggestion.

Reply:
Thanks @jrodewig and @debadair for the feedback!

In this particular case, the data is not necessarily deleted after the cold phase, it could be after the hot or warm depending on which phase is turned on. I was looking for a way to expressing that after all phases data will be deleted. Drawing from your recommendations, what do you think about:

"Policy deletes the index after all lifecycle phases"